core/controllers/classifier.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for communicating with the VM for training classifiers."""
+
+import hashlib
+import hmac
+import json
+
+from core.controllers import base
+from core.domain import classifier_domain
+from core.domain import classifier_services
+from core.domain import config_domain
+
+import feconf
+
+
+def verify_signature(message, vm_id, received_signature):
+    """Function that checks if the signature received from the VM is valid.
+
+    Args:
+        message: str. The string encoding of the dict that contains job_id and
+            classifier_data.
+        vm_id: str. The ID of the VM instance.
+        received_signature: str. The signature received from the VM.
+
+    Returns:
+        bool.
+
+    Raises:
+        UnauthorizedUserException.
+    """
+    secret = None
+    for val in config_domain.VMID_SHARED_SECRET_KEY_MAPPING.value:
+        if val['vm_id'] == vm_id:
+            secret = val['shared_secret_key'][0]
+            break
+    if secret is None:
+        return False
+    secret = str([val['shared_secret_key'] for val in (
+        config_domain.VMID_SHARED_SECRET_KEY_MAPPING.value) if val[
+            'vm_id'] == vm_id][0])
+    generated_signature = hmac.new(
```

Comment:
Done. @seanlip I hae a question. Can I import core.controllers.classifier in classifier_test and use the same generate_signature function? I wasn't sure since I've never seen it being done.